src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -1,93 +1,24 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System.Composition;
-using System.Linq;
-using System.Threading;
 using Microsoft.CodeAnalysis.CodeRefactorings;
 using Microsoft.CodeAnalysis.CodeRefactorings.InvertIf;
-using Microsoft.CodeAnalysis.CSharp.Extensions;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Microsoft.CodeAnalysis.Editing;
-using Microsoft.CodeAnalysis.Formatting;
-using Microsoft.CodeAnalysis.LanguageServices;
-using Microsoft.CodeAnalysis.Operations;
 using Microsoft.CodeAnalysis.Shared.Extensions;
-using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.InvertIf
 {
     [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.InvertIf), Shared]
-    internal partial class CSharpInvertIfCodeRefactoringProvider : AbstractInvertIfCodeRefactoringProvider
+    internal sealed partial class CSharpInvertIfCodeRefactoringProvider : AbstractInvertIfCodeRefactoringProvider
     {
-        protected override SyntaxNode GetIfStatement(TextSpan textSpan, SyntaxToken token, CancellationToken cancellationToken)
+        protected override SyntaxNode GetIfStatement(SyntaxToken token)
         {
-            var ifStatement = token.GetAncestor<IfStatementSyntax>();
-            if (ifStatement == null || ifStatement.Else == null)
-            {
-                return null;
-            }
-
-            var span = TextSpan.FromBounds(ifStatement.IfKeyword.Span.Start, ifStatement.CloseParenToken.Span.End);
-            if (!span.IntersectsWith(textSpan))
-            {
-                return null;
-            }
-
-            return ifStatement;
+            return token.GetAncestor<IfStatementSyntax>();
```

Comment:
nit =>

Reply:
Is the VB form much different?  this feels like the base class could do this work...

Reply:
yes

Reply:
how id teh VB form much different? just so i can understand?  is it that it supports multiline/singleline ifs?  if so, i think that should be two spearate providers.  which would make sense anyways as you have two analyzers...

Reply:
(resolved)